### PR TITLE
My Home: Remove "Visit Site" Button

### LINKED ISSUE #52809: Remove My Home "Visit site" button
Part of My Home Pod: Cleaning up My home
Project thread: pd2qGl-17-p2

The visit site action also exists in the side nav and thus is being removed from My Home in efforts to declutter the page.

### PR DESCRIPTION
**Pull request by Aurorum**
#### Changes proposed in this Pull Request

* Removes the "Visit Site" button on My Home and all associated styling

#### Testing instructions

Visit the section and verify that the button no longer appears (it used to be at the top right).

Screenshot 2021-05-13 at 16 47 11

Fixes #52809

### CODE REVIEW

**Review by sixhours — APPROVED** [review]
This is working for me! I will leave it to someone from the My Home group to give final approval. :)

File: client/my-sites/customer-home/main.jsx [hunk 1]
```diff
@@ -130,51 +115,32 @@ const Home = ( {
...
-	const { isStaticHomePage } = stateProps;
 	return {
 		...ownProps,
 		...stateProps,
-		trackViewSiteAction: () => dispatchProps.trackViewSiteAction( isStaticHomePage ),
 	};
 };
 
-const connectHome = connect( mapStateToProps, mapDispatchToProps, mergeProps );
+const connectHome = connect( mapStateToProps, null, mergeProps );
```
Comment (p-jackson) [inline client/my-sites/customer-home/main.jsx]: With `trackViewSiteAction` going away I think we can actually get rid of all this prop merging code! Looks like it was only there so we could forward the `isStaticHomePage` prop to the `trackViewSiteAction()` function.

```suggestion
const connectHome = connect( mapStateToProps );
```

And then we can delete `mergeProps` entirely

File: client/my-sites/customer-home/style.scss [hunk 2]
```diff
@@ -19,32 +19,6 @@
...
-	@include breakpoint-deprecated( '<660px' ) {
-		padding-right: 16px;
-	}
-
-	.formatted-header {
-		margin-right: 12px;
-	}
-
-	.formatted-header__subtitle {
-		margin: 0;
```
Comment (p-jackson) [inline client/my-sites/customer-home/style.scss:34]: Removing this `margin: 0` rule has changed the padding a little. Here's a gif of me switching tabs between your change and production:

May-14-2021 16-22-30

I think if we reinstate the `customer-home__heading` wrapper div, and this `formatted-header__subtitle` rule, then it'll keep the spacing exactly as it was before. Looks like all the rest is good to delete though 👍

**Review by p-jackson — DISMISSED** [review]
Thanks for all your help! It's amazing how much code can be cleaned up just removing this one button 🙂 

This diff introduces a small padding change, I've left some notes about how that might be fixed. Otherwise looking good 👍

File: client/my-sites/customer-home/main.jsx (hunk 1 above)
Comment (Aurorum) [reply client/my-sites/customer-home/main.jsx]: Nice spot - thank you! Should be removed now. :)

File: client/my-sites/customer-home/style.scss (hunk 2 above)
Comment (Aurorum) [reply client/my-sites/customer-home/style.scss:34]: Oops, should be fixed now! :)

**Review by roo2 — APPROVED** [review]
LGTM, great that we were able to remove so much code for one button

**Comment by autumnfjeld** [comment]
We should follow up by removing or update the metric events:

```
	composeAnalytics(
		recordTracksEvent( 'calypso_customer_home_my_site_view_site_click', {
			is_static_home_page: isStaticHomePage,
		} ),
		bumpStat( 'calypso_customer_home', 'my_site_view_site' )
	);
```

e.g.  update the description in tracks as removed or delete the event.

**Comment by jeffgolenski** [comment]
@p-jackson @autumnfjeld @roo2 - sorry friends, can we get this put back on my home per Dave's request until we can solidify some metrics around its usage? Thank you!

**Comment by Aurorum** [comment]
PR in #53078 to revert this. :) 

Screenshot 2021-05-21 at 15 03 47

**Comment by p-jackson** [comment]
No problem.
@jeffgolenski @davemart-in did you see this post that included some metrics? pd2qGl-3A-p2

**Pull request merged**